.github/workflows/cd-image.yaml

```diff
@@ -71,3 +77,22 @@ jobs:
             .
 
           echo "Used tags: $TAGS" >> $GITHUB_STEP_SUMMARY
+          echo "deploy_tag=${DEPLOY_TAG}" >> $GITHUB_OUTPUT
+
+      - name: Set outputs
+        id: set-outputs
+        run: |
+          echo "image_tag=${{ steps.build-push.outputs.deploy_tag }}" >> $GITHUB_OUTPUT
+
+  deploy-to-ci-cluster:
+    needs: publish-image-to-ecr
+    runs-on: ubuntu-latest
+    if: github.ref_name == 'main'
+    steps:
+      - name: Deploy image to nildb-ci-n1, nildb-ci-n2, nildb-ci-n3
+        uses: peter-evans/repository-dispatch@v2
+        with:
+          token: ${{ secrets.PAT_TOKEN }}
+          repository: NillionNetwork/nildb-infra
+          event-type: trigger-on-new-nildb-image
+          client-payload: '{"ref": "${{ github.ref }}", "sha": "${{ github.sha }}", "tag": "${{ needs.publish-image-to-ecr.outputs.image_tag }}"}'
```

Comment:
would you mind elaborating on what this action is doing and why it requires a PAT?

Reply:
The aim is to invoke the `ci-n1,2,3` workflows in nildb-infra, for example the `event-type` maps to [here](https://github.com/NillionNetwork/nildb-infra/blob/ccf5867b787e1227368acf8479d9e6beb99b8e15/.github/workflows/ecs-ci-n1.yaml#L5). Since the repo is private it needs an access token.

That said, i should convert this to a curl request so that we don't need to use the external library. Let me convert this back to draft and i'll do some more iterations tomorrow.